Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -195,214 +255,169 @@ float4 SharpBilinearSample(float3 uvw, float gamma)
+    else if (resampling_method == 3)
+        color = BicubicSample(uvw, CUBIC_COEFF_GEN(1.0 / 3.0, 1.0 / 3.0));
+
+    // Bicubic: Catmull-Rom
+    else if (resampling_method == 4)
+        color = BicubicSample(uvw, CUBIC_COEFF_GEN(0.0, 0.5));
+
+    // Sharp Bilinear
+    else if (resampling_method == 5)
+        color = SharpBilinearSample(uvw);
+
+    // Area Sampling
+    else if (resampling_method == 6)
+        color = AreaSampling(uvw);
+
+    // Nearest Neighbor
+    else if (resampling_method == 7)
+        color = QuickSample(uvw);
+
+    // Bicubic: Hermite
+    else if (resampling_method == 8)
+        color = BicubicSample(uvw, CUBIC_COEFF_GEN(0.0, 0.0));
+
+    return color;
 }
 
 void main()
 {
-	// This tries to fall back on GPU HW sampling if it can (it won't be gamma corrected).
-	bool raw_resampling = resampling_method <= 0;
-	bool needs_rescaling = GetResolution() != GetWindowResolution();
-
-	bool needs_resampling = needs_rescaling && (OptionEnabled(hdr_output) || OptionEnabled(correct_gamma) || !raw_resampling);
-
-	float4 color;
-
-	if (needs_resampling)
-	{
-		// Doing linear sampling in "gamma space" on linear texture formats isn't correct.
-		// If the source and target resolutions don't match, the GPU will return a color
-		// that is the average of 4 gamma space colors, but gamma space colors can't be blended together,
-		// gamma neeeds to be de-applied first. This makes a big difference if colors change
-		// drastically between two pixels.
-
-		color = LinearGammaCorrectedSample(game_gamma);
-	}
-	else
-	{
-		// Default GPU HW sampling. Bilinear is identical to Nearest Neighbor if the input and output resolutions match.
-		if (needs_rescaling)
-			color = texture(samp0, v_tex0);
-		else
-			color = texture(samp1, v_tex0);
-
-		// Convert to linear before doing any other of follow up operations.
-		color.rgb = pow(color.rgb, float3(game_gamma));
-	}
-
-	if (OptionEnabled(correct_color_space))
-	{
-		if (game_color_space == 0)
-			color.rgb = color.rgb * from_NTSCM;
-		else if (game_color_space == 1)
-			color.rgb = color.rgb * from_NTSCJ;
-		else if (game_color_space == 2)
-			color.rgb = color.rgb * from_PAL;
-	}
-
-	if (OptionEnabled(hdr_output))
-	{
-		float hdr_paper_white = hdr_paper_white_nits / hdr_sdr_white_nits;
-		color.rgb *= hdr_paper_white;
-	}
-
-	if (OptionEnabled(linear_space_output))
-	{
-		// Nothing to do here
-	}
-	// Correct the SDR gamma for sRGB (PC/Monitor) or ~2.2 (Common TV gamma)
-	else if (OptionEnabled(correct_gamma))
-	{
-		if (OptionEnabled(sdr_display_gamma_sRGB))
-			color.rgb = LinearTosRGBGamma(color.rgb);
-		else
-			color.rgb = pow(color.rgb, float3(1.0 / sdr_display_custom_gamma));
-	}
-	// Restore the original gamma without changes
-	else
-	{
-			color.rgb = pow(color.rgb, float3(1.0 / game_gamma));
-	}
-
-	SetOutput(color);
+    float4 color = float4(0.0, 0.0, 0.0, 1.0);
+
+    if (GetResolution() == GetWindowResolution() || resampling_method == 0)
```

Comment:
This new code removes all the comments that explained the logic of what was being done.
This branch was checking for `needs_resampling`:
```
bool raw_resampling = resampling_method <= 0;
bool needs_rescaling = GetResolution() != GetWindowResolution();

bool needs_resampling = needs_rescaling && (OptionEnabled(hdr_output) || OptionEnabled(correct_gamma) || !raw_resampling);
```
while now it does some inline checks that are not clearly why they are being made.